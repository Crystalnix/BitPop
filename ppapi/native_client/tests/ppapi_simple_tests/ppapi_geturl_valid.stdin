# define a bunch of variables for readability

set_variable MODULE i(4444)
set_variable INSTANCE i(5555)

# Tags used by pepper/srpc to marshall js parameters
set_variable JS_STR 4:5
set_variable JS_INT 4:2

# NOTE: the strings below must not contain unescaped spaces or commas
set_variable ARG_loadUrl ${JS_STR},4:7,"loadUrl",1:0
set_variable ARG_ReportResult ${JS_STR},4:12,"ReportResult",4:0

set_variable ARG_URL ${JS_STR},4:25,"ppapi_geturl_success.html",7:0
set_variable ARG_URL_FULL ${JS_STR},4:47,"http://localhost:5103/ppapi_geturl_success.html",1:0

set_variable ARG_GET  ${JS_STR},4:3,"GET",5:0

set_variable ARG_SUCC ${JS_STR},4:12,"hello\x20world!",4:0
set_variable ARG_0 ${JS_INT},4:0
set_variable ARG_1 ${JS_INT},4:1

set_variable WINDOW R(24,8:6,8:0x9f,8:0x5f29)
# TODO(robertm) figure out what this represents, maybe PP_Var
set_variable EXCEPTION R(8,8:0)
set_variable NULL R(8,8:0)

echo
echo "*** INIT MODULE"
replay 99 PPB_GetInterface:s:i  s("PPB_Core;1.0") * i(1)
replay 99 PPB_GetInterface:s:i  s("PPB_Instance;1.0") * i(1)
replay 99 PPB_GetInterface:s:i  s("PPB_Var;1.0") * i(1)
# the results contain a process id which is non-determinisitic
rpc PPP_InitializeModule hide-results i(0) ${MODULE} h(pepper_desc) s("${service_string}") * i(0) i(0)

echo
echo "*** TRIGGER REPAINT"
rpc PPP_Instance_DidChangeView ${INSTANCE} I(4,8,79,400,400) I(4,0,0,400,400) *
# leave time for flush of stdout

echo
echo "*** GET INSTANCE OBJECT"
rpc PPP_Instance_GetInstanceObject hide-results ${INSTANCE} * C(16)
# save result to variable and also register it as non-deterministic
set_variable OBJECT ${result0}
nondeterministic ${result0} @THE_OBJECT

######################################################################
## URL (buffer)
######################################################################

echo
echo "*** CALL loadUrl(): as buffer"
replay 1 PPB_GetInterface:s:i  s("PPB_URLRequestInfo;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLResponseInfo;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLLoader;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_FileIO;1.0") * i(1)

replay 1 PPB_FileIO_Create:i:i ${INSTANCE} * i(35)
replay 1 PPB_URLRequestInfo_Create:i:i ${INSTANCE} * i(30)
replay 1 PPB_URLLoader_Create:i:i  ${INSTANCE} * i(34)

replay 1 PPB_FileIO_IsFileIO:i:i i(35) * i(1)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(35) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(35) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(35) * i(0)

replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(30) * i(1)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(30) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(30) * i(0)
replay 1 PPB_FileIO_IsFileIO:i:i i(30) * i(0)

replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(34) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(34) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(34) * i(1)
replay 1 PPB_FileIO_IsFileIO:i:i i(34) * i(0)

replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(0) R(40,${ARG_URL}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(1) R(16,${ARG_GET}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(3) R(8,${ARG_0}) * i(1)

replay 1 PPB_URLLoader_Open:iii:i i(34) i(30) i(1) * i(-1)
rpc Call ${OBJECT} R(16,${ARG_loadUrl}) i(2) R(48,${ARG_URL},${ARG_0}) ${NULL} * C(8) C(8)

echo
echo "*** async response part1"
replay 1 PPB_URLLoader_GetResponseInfo:i:i i(34) * i(38)
replay 1 PPB_FileIO_IsFileIO:i:i i(38) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(38) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(38) * i(1)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(38) * i(0)
replay 1 PPB_URLResponseInfo_GetProperty:ii:C i(38) i(0) * R(56,${ARG_URL_FULL})
replay 1 PPB_URLResponseInfo_GetProperty:ii:C i(38) i(3) * R(8,4:3,4:200)
replay 1 PPB_URLLoader_ReadResponseBody:iii:Ci i(34) i(1024) i(2) * C(0,) i(-1)
rpc RunCompletionCallback i(1) i(0) C(0,) *

echo
echo "*** async response part2"
replay 1 PPB_URLLoader_ReadResponseBody:iii:Ci i(34) i(1024) i(3) * C(0,) i(-1)
rpc RunCompletionCallback i(2) i(12) C(12,hello\x20world!) *

######################################################################
## URL (file)
######################################################################

echo
echo "*** CALL loadUrl(): as file"
replay 1 PPB_GetInterface:s:i  s("PPB_FileIO;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLRequestInfo;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLResponseInfo;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLLoader;1.0") * i(1)

replay 1 PPB_URLLoader_Create:i:i  ${INSTANCE} * i(34)
replay 1 PPB_URLRequestInfo_Create:i:i ${INSTANCE} * i(30)
replay 1 PPB_FileIO_Create:i:i ${INSTANCE} * i(66)

replay 1 PPB_FileIO_IsFileIO:i:i i(66) * i(1)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(66) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(66) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(66) * i(0)

replay 1 PPB_FileIO_IsFileIO:i:i i(30) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(30) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(30) * i(1)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(30) * i(0)

replay 1 PPB_FileIO_IsFileIO:i:i i(34) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(34) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(34) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(34) * i(1)

replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(0) R(40,${ARG_URL}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(1) R(16,${ARG_GET}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(3) R(8,${ARG_1}) * i(1)

replay 1 PPB_URLLoader_Open:iii:i i(34) i(30) i(4) * i(-1)

replay 1 PPB_Core_ReleaseResource:i: i(30) *
replay 1 PPB_Core_ReleaseResource:i: i(34) *
replay 1 PPB_Core_ReleaseResource:i: i(66) *

replay 1 Call:CCiCC:CC R(16,8:0x9f,8:0x5f29) R(24,${ARG_ReportResult}) i(4) R(80,${ARG_URL},${ARG_1},${ARG_SUCC},${ARG_1}) ${NULL} * ${NULL} ${NULL}

rpc Call ${OBJECT} R(16,${ARG_loadUrl}) i(2) R(48,${ARG_URL},${ARG_1}) ${NULL} * C(8) C(8)

######################################################################
## EPILOG
######################################################################
echo
echo "*** DONE"
rpc PPP_ShutdownModule *
