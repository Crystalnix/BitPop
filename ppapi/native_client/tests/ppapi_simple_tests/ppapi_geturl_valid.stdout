
"*** REPLAY PROLOG"

"*** REGISTER REPLAY METHODS"

"*** INITIALIZE PEPPER"

"*** CREATE UPCALL SERVICE"

"*** INIT MODULE"
rpc call intiated PPP_InitializeModule:iihs:ii
input 0:  i(0)
input 1:  i(4444)
input 2:  h(pepper_desc)
input 3:  @LENGTHY_SERIVE_STRING
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_Core;1.0")
output 0:  i(1)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_Instance;1.0")
output 0:  i(1)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_Var;1.0")
output 0:  i(1)
rpc call complete PPP_InitializeModule:iihs:ii


"*** TRIGGER REPAINT"
rpc call intiated PPP_Instance_DidChangeView:iII:
input 0:  i(5555)
input 1:  I(4,8,79,400,400)
input 2:  I(4,0,0,400,400)
rpc call complete PPP_Instance_DidChangeView:iII:


"*** CALL loadUrl(): as buffer"
rpc call intiated Call:CCiCC:CC
input 0:  @THE_OBJECT
input 1:  C(16,\x05\x00\x00\x00\x07\x00\x00\x00loadUrl\x00)
input 2:  i(2)
input 3:  C(48,\x05\x00\x00\x00\x19\x00\x00\x00ppapi_geturl_success.html\x00\x00\x00\x00\x00\x00\x00\x02\x00\x00\x00\x00\x00\x00\x00)
input 4:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_URLRequestInfo;1.0")
output 0:  i(1)
replaying PPB_URLRequestInfo_Create:i:i:
input 0:  i(5555)
output 0:  i(30)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_URLResponseInfo;1.0")
output 0:  i(1)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_URLLoader;1.0")
output 0:  i(1)
replaying PPB_URLLoader_Create:i:i:
input 0:  i(5555)
output 0:  i(34)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_FileIO;1.0")
output 0:  i(1)
replaying PPB_FileIO_Create:i:i:
input 0:  i(5555)
output 0:  i(35)
replaying PPB_FileIO_IsFileIO:i:i:
input 0:  i(30)
output 0:  i(0)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
input 0:  i(30)
output 0:  i(1)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
input 0:  i(30)
output 0:  i(0)
replaying PPB_URLLoader_IsURLLoader:i:i:
input 0:  i(30)
output 0:  i(0)
replaying PPB_FileIO_IsFileIO:i:i:
input 0:  i(34)
output 0:  i(0)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
input 0:  i(34)
output 0:  i(0)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
input 0:  i(34)
output 0:  i(0)
replaying PPB_URLLoader_IsURLLoader:i:i:
input 0:  i(34)
output 0:  i(1)
replaying PPB_FileIO_IsFileIO:i:i:
input 0:  i(35)
output 0:  i(1)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
input 0:  i(35)
output 0:  i(0)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
input 0:  i(35)
output 0:  i(0)
replaying PPB_URLLoader_IsURLLoader:i:i:
input 0:  i(35)
output 0:  i(0)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
input 0:  i(30)
input 1:  i(0)
input 2:  C(40,\x05\x00\x00\x00\x19\x00\x00\x00ppapi_geturl_success.html\x00\x00\x00\x00\x00\x00\x00)
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
input 0:  i(30)
input 1:  i(1)
input 2:  C(16,\x05\x00\x00\x00\x03\x00\x00\x00GET\x00\x00\x00\x00\x00)
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
input 0:  i(30)
input 1:  i(3)
input 2:  C(8,\x02\x00\x00\x00\x00\x00\x00\x00)
output 0:  i(1)
replaying PPB_URLLoader_Open:iii:i:
input 0:  i(34)
input 1:  i(30)
input 2:  i(1)
output 0:  i(-1)
rpc call complete Call:CCiCC:CC
output 0:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)
output 1:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)


"*** async response part1"
rpc call intiated RunCompletionCallback:iiC:
input 0:  i(1)
input 1:  i(0)
input 2:  C(0,)
replaying PPB_URLLoader_GetResponseInfo:i:i:
input 0:  i(34)
output 0:  i(38)
replaying PPB_FileIO_IsFileIO:i:i:
input 0:  i(38)
output 0:  i(0)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
input 0:  i(38)
output 0:  i(0)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
input 0:  i(38)
output 0:  i(1)
replaying PPB_URLLoader_IsURLLoader:i:i:
input 0:  i(38)
output 0:  i(0)
replaying PPB_URLResponseInfo_GetProperty:ii:C:
input 0:  i(38)
input 1:  i(0)
output 0:  C(56,\x05\x00\x00\x00/\x00\x00\x00http://localhost:5103/ppapi_geturl_success.html\x00)
replaying PPB_URLResponseInfo_GetProperty:ii:C:
input 0:  i(38)
input 1:  i(3)
output 0:  C(8,\x03\x00\x00\x00\xc8\x00\x00\x00)
replaying PPB_URLLoader_ReadResponseBody:iii:Ci:
input 0:  i(34)
input 1:  i(1024)
input 2:  i(2)
output 0:  C(0,)
output 1:  i(-1)
rpc call complete RunCompletionCallback:iiC:


"*** async response part2"
rpc call intiated RunCompletionCallback:iiC:
input 0:  i(2)
input 1:  i(12)
input 2:  C(12,hello world!)
replaying PPB_URLLoader_ReadResponseBody:iii:Ci:
input 0:  i(34)
input 1:  i(1024)
input 2:  i(3)
output 0:  C(0,)
output 1:  i(-1)
rpc call complete RunCompletionCallback:iiC:


"*** async response eof - report success"
rpc call intiated RunCompletionCallback:iiC:
input 0:  i(3)
input 1:  i(0)
input 2:  C(0,)
replaying PPB_Core_ReleaseResource:i::
input 0:  i(30)
replaying PPB_Core_ReleaseResource:i::
input 0:  i(34)
replaying PPB_Core_ReleaseResource:i::
input 0:  i(38)
replaying PPB_Core_ReleaseResource:i::
input 0:  i(35)
rpc call complete RunCompletionCallback:iiC:


"*** CALL loadUrl(): as file"
rpc call intiated Call:CCiCC:CC
input 0:  @THE_OBJECT
input 1:  C(16,\x05\x00\x00\x00\x07\x00\x00\x00loadUrl\x00)
input 2:  i(2)
input 3:  C(48,\x05\x00\x00\x00\x19\x00\x00\x00ppapi_geturl_success.html\x00\x00\x00\x00\x00\x00\x00\x02\x00\x00\x00\x01\x00\x00\x00)
input 4:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)
replaying PPB_URLRequestInfo_Create:i:i:
input 0:  i(5555)
output 0:  i(30)
replaying PPB_URLLoader_Create:i:i:
input 0:  i(5555)
output 0:  i(34)
replaying PPB_FileIO_Create:i:i:
input 0:  i(5555)
output 0:  i(66)
replaying PPB_FileIO_IsFileIO:i:i:
input 0:  i(30)
output 0:  i(0)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
input 0:  i(30)
output 0:  i(1)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
input 0:  i(30)
output 0:  i(0)
replaying PPB_URLLoader_IsURLLoader:i:i:
input 0:  i(30)
output 0:  i(0)
replaying PPB_FileIO_IsFileIO:i:i:
input 0:  i(34)
output 0:  i(0)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
input 0:  i(34)
output 0:  i(0)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
input 0:  i(34)
output 0:  i(0)
replaying PPB_URLLoader_IsURLLoader:i:i:
input 0:  i(34)
output 0:  i(1)
replaying PPB_FileIO_IsFileIO:i:i:
input 0:  i(66)
output 0:  i(1)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
input 0:  i(66)
output 0:  i(0)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
input 0:  i(66)
output 0:  i(0)
replaying PPB_URLLoader_IsURLLoader:i:i:
input 0:  i(66)
output 0:  i(0)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
input 0:  i(30)
input 1:  i(0)
input 2:  C(40,\x05\x00\x00\x00\x19\x00\x00\x00ppapi_geturl_success.html\x00\x00\x00\x00\x00\x00\x00)
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
input 0:  i(30)
input 1:  i(1)
input 2:  C(16,\x05\x00\x00\x00\x03\x00\x00\x00GET\x00\x00\x00\x00\x00)
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
input 0:  i(30)
input 1:  i(3)
input 2:  C(8,\x02\x00\x00\x00\x01\x00\x00\x00)
output 0:  i(1)
replaying PPB_URLLoader_Open:iii:i:
input 0:  i(34)
input 1:  i(30)
input 2:  i(4)
output 0:  i(-1)
rpc call complete Call:CCiCC:CC
output 0:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)
output 1:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)


"*** DONE"
rpc call intiated PPP_ShutdownModule::
rpc call complete PPP_ShutdownModule::

